ldawithgridsearch.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import pandas as pd
+import re, nltk, spacy, gensim
+# SKLearn
+from sklearn.decomposition import LatentDirichletAllocation, TruncatedSVD
+from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
+from sklearn.model_selection import GridSearchCV
+
+dtm = pd.read_pickle("dtm.pkl")
+
+
+# Build LDA Model
+def lda_grid_search(n1, n2, n3, d1, d2, d3):
+    # Define Search Param
+    search_params = {'n_components': [n1, n2, n3], 'learning_decay': [d1, d2, d3]}
+    # Init the Model
+    lda = LatentDirichletAllocation()
+    # Init Grid Search Class
+    model = GridSearchCV(lda, param_grid=search_params)
+    # Do the Grid Search
+    model.fit(dtm)
+    # Model Parameters
+    print("Best Model's Params: ", model.best_params_)
+    # Log Likelihood Score
+    print("Best Log Likelihood Score: ", model.best_score_)
+    # Perplexity
+    print("Model Perplexity: ", model.best_estimator_.perplexity(dtm))
```

Comment:
Print statements are good to debug your code and understand what it is doing, but do make sure they don't make their way to the production code.